pythran/pythonic/include/numpy/fft/fft.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef PYTHONIC_INCLUDE_NUMPY_FFT_FFT_HPP
+#define PYTHONIC_INCLUDE_NUMPY_FFT_FFT_HPP
+
+#include "pythonic/include/utils/functor.hpp"
+#include "pythonic/include/types/ndarray.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+  namespace fft
+  {
+
+    template <class T, class pS>
+    types::ndarray<std::complex<T>,
+            types::array<long, std::tuple_size<pS>::value>>
+    fft(types::ndarray<std::complex<T>, pS> const &a, long n = -1, long axis = -1,
+          types::str const &norm = {});
+
+    template <class T, class pS>
+    types::ndarray<std::complex<T>,
+            types::array<long, std::tuple_size<pS>::value>>
+    fft(types::ndarray<std::complex<T>, pS> const &a, types::none_type n, long axis,
+          types::str const &norm );
+
+    template <class T, class pS>
+    types::ndarray<std::complex<T>,
+            types::array<long, std::tuple_size<pS>::value>>
+    fft(types::ndarray<std::complex<T>, pS> const &a, long n, long axis,
+          types::none_type norm );
+
+    template <class T, class pS>
+    types::ndarray<std::complex<T>,
+            types::array<long, std::tuple_size<pS>::value>>
+    fft(types::ndarray<std::complex<T>, pS> const &a, types::none_type n , long axis =-1,
+          types::none_type norm = types::none_type{});
+
+    NUMPY_EXPR_TO_NDARRAY0_DECL(fft);
+    DEFINE_FUNCTOR(pythonic::numpy::fft, fft);
+  }
```

Comment:
this should work for non-complex floats too right?

Reply:
How do I convert an ndarray? I've tried using astype, but am failing.

Reply:
Fixed, fft and ifft now work for non-complex inputs.

Reply:
I needed to include quite a few overloaded functions, maybe there is a more elegant way of doing this, however I could not find one.

Reply:
Sorry I should have said it works for floating point inputs. I need to look into what  numpy does for integer inputs (It seems it's not just a conversion to complex.), but I would argue that integer inputs is quite uncommon and can maybe put off to another merge request